### PR TITLE
feat: create automatic setup script for the project devcontainers

### PR DESCRIPTION
**Pull request by joaquimds**
## Description
Adds an automatic setup script and simplifies the README as part of improving the Mapped dev environment.

## Motivation and Context
https://linear.app/commonknowledge/issue/MAP-505/everyone-in-the-co-op-should-be-able-to-run-an-identical-instance-of

## How Can It Be Tested?
Destroy your current Mapped dev environment. In the project directory:
```
docker compose down -v
rm .env
rm data/2024_general_election.csv
rm -rf .venv
rm -rf nextjs/node_modules
rm -rf nextjs/.next
rm -rf nextjs/.env
```

Then follow the steps in `README.md` to set up the new environment, and go through the feature showcase (also in the readme).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,99 +2,62 @@
...
 
-You could alternatively run commands individually (again, from inside the container), eg:
+6. Visit http://localhost:3000 in your browser to access Mapped.
 
-    ./manage.py import_areas
-    ./manage.py import_mps
+### Feature Showcase
 
-### Logging In
+1. Log in to the Mapped front end.
```
[the remark this answers is not in the document]
Comment (joaquimds) [reply README.md:44]: I don't think there should be instructions on how to get to the login page, it should be obvious from the UX of the site. If it's not then that's a bigger problem for us.

File: README.md [hunk 2]
```diff
@@ -2,99 +2,59 @@
...
-    docker-compose exec web bash
-
-If you’re running Docker via Visual Studio Code, instead, you’ll want to run the commands via the built-in terminal.
-
-You will likely want to create a Django superuser, by running this inside the container:
-
-    script/createsuperuser
-
-The superuser will be created with the details specified in the `DJANGO_SUPERUSER_*` environment variables. [Read more about how Docker handles environment variables](https://docs.docker.com/compose/envvars-precedence/).
+```bash
```
[the remark this answers is not in the document]
Comment (joaquimds) [reply README.md:23]: I think in this instance, the benefit of combining 2 commands into 1 is outweight by the benefit of getting familiar with the `bw` CLI.

File: README.md [hunk 3]
```diff
@@ -2,99 +2,63 @@
...
+6. Visit http://localhost:3000 in your browser to access Mapped.
 
-    ./manage.py import_areas
-    ./manage.py import_mps
+### Feature Showcase
 
-### Logging In
+1. Log in to the Mapped [front end](http://localhost:3000/login). The username and password are `admin` and `password`.
+2. You should have 3 data sources: "Seed Member List", "Seed Custom Data", and "Seed Events".
+3. This data source should have a mapping that populates the Constituency and Element fields in the AirTable.
```
Comment (ev-sc) [inline README.md:46]: Which data source?

File: README.md [hunk 4]
```diff
@@ -2,99 +2,63 @@
...
-You will likely want to create a Django superuser, by running this inside the container:
-
-    script/createsuperuser
+```bash
+bw login
+bw get attachment .env --itemid 064abbb1-e41b-4632-be90-b2270105d4d1
+```
 
-The superuser will be created with the details specified in the `DJANGO_SUPERUSER_*` environment variables. [Read more about how Docker handles environment variables](https://docs.docker.com/compose/envvars-precedence/).
+2. Open the project folder in VSCode. You should receive a notification in the bottom left to open the project in
```
Comment (ev-sc) [inline README.md:28]: Please replace all incidences of "should" with definitives like "will", "is", "are" etc.
They're unnecessary; they leave a linguistic space for the possibility that something may not happen as intended, so priming the user. However, there's no function to this: if the expected thing doesn't happen, then the user knows something is wrong. The "should" doesn't help, it just seeds a background sense of uncertainty.

File: README.md [hunk 5]
```diff
@@ -2,99 +2,63 @@
...
-    ./manage.py import_mps
+### Feature Showcase
 
-### Logging In
+1. Log in to the Mapped [front end](http://localhost:3000/login). The username and password are `admin` and `password`.
+2. You should have 3 data sources: "Seed Member List", "Seed Custom Data", and "Seed Events".
+3. This data source should have a mapping that populates the Constituency and Element fields in the AirTable.
+4. You should have a map report called "Test map report". Constituencies and MPs should be displayed.
+5. There should be a hub at [http://hub.localhost:3000](http://hub.localhost:3000).
+6. The hub should have some minimal content on the homepage, and a working pledge map.
```
Comment (ev-sc) [inline README.md:49]: Clicking the "pledge to take part" button doesn't work:

image

File: README.md [hunk 6]
```diff
@@ -2,99 +2,63 @@
...
 
-    ./manage.py import_areas
-    ./manage.py import_mps
+### Feature Showcase
 
-### Logging In
+1. Log in to the Mapped [front end](http://localhost:3000/login). The username and password are `admin` and `password`.
+2. You should have 3 data sources: "Seed Member List", "Seed Custom Data", and "Seed Events".
+3. This data source should have a mapping that populates the Constituency and Element fields in the AirTable.
+4. You should have a map report called "Test map report". Constituencies and MPs should be displayed.
```
Comment (ev-sc) [inline README.md:47]: The map screen is blank and there are console errors:

image

File: README.md (hunk 4 above)
Comment (ev-sc) [reply README.md]: @joaquimds I've just made these changes myself to illustrate what I mean 😁

**Comment by joaquimds** [comment]
@ev-sc there was a missing .env file, so I've created the new script ./bin/download_env_files.sh to download both the scripts. If you pull this, then run it, the front-end maps should work.

File: README.md (hunk 5 above)
Comment (ev-sc) [reply README.md]: This is still an issue

Comment (ev-sc) [reply README.md]: I'm going to split this off into a new issue.

**Review by ev-sc — APPROVED** [review]

**Pull request merged**